map_gui/src/common/colors.rs

```diff
@@ -17,14 +17,14 @@ pub struct ColorDiscrete<'a> {
 }
 
 impl<'a> ColorDiscrete<'a> {
-    pub fn new(app: &'a App, categories: Vec<(&'static str, Color)>) -> ColorDiscrete<'a> {
+    pub fn new(app: &'a dyn AppLike, categories: Vec<(&'static str, Color)>) -> ColorDiscrete<'a> {
```

Comment:
huh, I didn't know you could use dye outside of a `Box`.

Reply:
I think it works only because we're not trying to store the trait object anywhere. If we were, we'd need to know the size. I guess the size of the pointer on the stack is fixed.